### PR TITLE
DTSPB-2683-2987-Amend-GOP-LegalStatement

### PR DESCRIPTION
**Pull request by FeliTam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-2683
https://tools.hmcts.net/jira/browse/DTSPB-2987

### Change description ###
Amend GOP Legal Statement wordings to react on willNumberOfCodicils
in both CW and Solicitor flow

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java [hunk 1]
```diff
@@ -1363,7 +1363,8 @@ private void updateCaseBuilder(CaseData caseData, ResponseCaseDataBuilder<?, ?>
 
         if (!isCodicil(caseData)) {
             builder
-                    .willNumberOfCodicils(null);
+                    .willNumberOfCodicils(null)
+                    .codicilAddedDateList(null);
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: I dont understand, why are we setting this list to null?

File: src/main/resources/templates/pdf/legalStatementProbate.html [hunk 2]
```diff
@@ -44,6 +44,7 @@
 </head>
 <body>
 <table>
+    {% set numberOfCodicils = case_details.case_data.codicilAddedDateList|length %}
```
Comment (sanjayparekh-hmcts) [inline src/main/resources/templates/pdf/legalStatementProbate.html:47]: we are probably better off aligning these HTML changes with what Iswarya has done for her JIRAs
I think she takes a null value into account
Please check with her

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java (hunk 1 above)
Comment (FeliTam) [reply src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: It is to fix a bug I found during testing.
Scenario:
User put "Yes" to "Were any codicils added", add codicil date and then Submit.
User amend detail again and put "No" to "Were any codicils added" without removing codicil date manually and then Submit.
Legal statement generated incorrectly as the codicil date list is not empty.

File: src/main/resources/templates/pdf/legalStatementProbate.html (hunk 2 above)
Comment (FeliTam) [reply src/main/resources/templates/pdf/legalStatementProbate.html:47]: Aligned template with changes made by Iswarya

**Pull request closed without merging**